core/commands/object/diff.go

```diff
@@ -0,0 +1,103 @@
+package objectcmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	cmds "github.com/ipfs/go-ipfs/commands"
+	dagutils "github.com/ipfs/go-ipfs/merkledag/utils"
+	path "github.com/ipfs/go-ipfs/path"
+)
+
+var ObjectDiffCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "takes a diff of the two given objects",
+		ShortDescription: `
+ipfs object diff is a command used to show the differences between
+two ipfs objects.
+
+Example:
```

Comment:
Nit: put the example in the `LongDescription` so that shorthelp only has the vitals.